src/lib/hooks/usePoll.ts

```diff
@@ -0,0 +1,57 @@
+import ms from 'ms.macro'
+import { useEffect, useMemo, useState } from 'react'
+
+const DEFAULT_POLLING_INTERVAL = ms`15s`
+const DEFAULT_KEEP_UNUSED_DATA_FOR = ms`10s`
+
+export default function usePoll<T>(
+  fetch: () => Promise<T>,
+  key = '',
+  pollingInterval = DEFAULT_POLLING_INTERVAL,
+  keepUnusedDataFor = DEFAULT_KEEP_UNUSED_DATA_FOR
+): T | undefined {
+  const cache = useMemo(() => new Map<string, { ttl: number; result?: T }>(), [])
+  const [data, setData] = useState<{ key: string; result?: T }>({ key })
+
+  useEffect(() => {
```

Comment:
can you add a comment above each useEffect to indicate what variable it tracks? The dependency array contains a few more objects that are used inside the body

Reply:
Added a comment to the other useEffect, and a `void key` to enforce it in code.
Did not add a comment to this one, as all tracked variables are used.